### PR TITLE
Model Optimizer detects when layout conversion for TF models should be disabled

### PR DESCRIPTION
**Pull request by lazarevevgeny**
Root cause analysis: Some TF models does not have Convolution operations with inputs in NHWC layout. For these models users must specify `\-\-disable_nhwc_to_nchw` command line parameter to disable model conversion from NHWC to NCHW layout. If user does not specify it then the model will be converted incorrectly.

Solution: Add code which traverses the model and looks for convolution operations in NHWC layout. If these operations are not found then the model is considered in NCHW layout. However, user can force conversion of model layout using new command line option `\-\-force_nhwc_to_nchw`.

MR number for the product-config repo is 1879.
MR number for the e2e test changes is 1027.

Ticket: 51259

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR: N/A no new transformation
* [x]  Transformation preserves original framework node names: N/A no new transformations

Validation:
* [x]  Unit tests
* [x]  Framework operation tests: N/A no new 
* [x]  Transformation tests: added
* [x]  Model Optimizer IR Reader check: Manually done

Documentation:
* [x]  Supported frameworks operations list: N/A since no new operations enabled
* [x]  Guide on how to convert the **public** model: Done
* [x]  User guide update: Done

### CODE REVIEW

File: model-optimizer/extensions/load/tf/loader.py [hunk 1]
```diff
@@ -100,5 +101,64 @@ def load(self, graph: Graph):
...
+    :return: Boolean result of the check
+    """
+    NHWC_conv_detected = False
+    for node in graph.get_op_nodes():
+        if is_node_layout_nhwc(node):
+            NHWC_conv_detected = True
+            break
+
+        # for the Loop node we need to check that the body does not contain marker ops as well
+        if node.op == 'Loop':
```
Comment (evolosen) [inline model-optimizer/extensions/load/tf/loader.py:150]: use node.sofr_get('op')

Comment (evolosen) [reply model-optimizer/extensions/load/tf/loader.py:150]: we need add check for If operation? if yes, please add comment about if.

Comment (lazarevevgeny) [reply model-optimizer/extensions/load/tf/loader.py:150]: 'op' attribute always exists, so there is no need for `soft_get`

Comment (lazarevevgeny) [reply model-optimizer/extensions/load/tf/loader.py:150]: Comment will be added with the next code update

**Review by achetver — APPROVED** [review]

**Comment by achetver** [comment]
In general looks good.

**Review by evolosen — APPROVED** [review]

**Comment by lazarevevgeny** [comment]
The changes are not needed anymore due to new API planned.

**Pull request closed without merging**